### PR TITLE
Create helper-update-manifest-readme

### PR DESCRIPTION
**Pull request by Thatoo**
Retrieve sources, get sha256sum, parse package version and update these strings in manifest, README and conf/app.src

### CODE REVIEW

**Comment by Thatoo** [comment]
I'm quiet confident about CHECKSUM and the sed for SOURCE_SUM.
However, I don't know how to parse tag_name from https://api.github.com/repos/duniter/cesium/releases/latest to make PACKAGE_VERSION equal to tag_name. I'm not sure about my other sed command as they have plenty of * and " in strings... I got confused.

**Comment by Thatoo** [comment]
PACKAGE_VERSION=$parse tag_name https://api.github.com/repos/duniter/cesium/releases/latest
or 
PACKAGE_VERSION=$cat https://api.github.com/repos/duniter/cesium/releases/latest | grep tag_name
is there not something to do that?

**Comment by Thatoo** [comment]
OK, I've managed to create all sed command.
All command works as desired and I can execute them one after the other to get the result I want.
However, I can't execute this script in my computer.
Is there a way to make it executed automatically?

**Comment by Thatoo** [comment]
Oh, and did you get the Pull Request from the command request-pull I did once everything was working?

**Comment by Thatoo** [comment]
I guess there is something wrong with my command
git push
git request-pull
I guess it has something to do with the fact I'm doing it from the branch Thatoo-test...
@M5oul and @kay0u : what do you think about it?

File: scripts/helper-update-version [hunk 1]
```diff
@@ -0,0 +1,17 @@
+ #!/bin/bash
+ # Retrieve sources, get sha256sum, parse package version and update these strings in manifest, README and /conf/app.src
+ 
+url=$(curl -s https://api.github.com/repos/duniter/cesium/releases/latest | grep "browser_" | grep "web" | head -1 | cut -d\" -f4)
+wget -nc --quiet $url -P ./tmp
+CHECKSUM=$(sudo sha256sum ./tmp/cesium-*-web.zip | head -c 64)
```
Comment (M5oul) [inline scripts/helper-update-version]: Why do you need a `sudo` here?

File: scripts/helper-update-version [hunk 2]
```diff
@@ -0,0 +1,17 @@
+ #!/bin/bash
```
Comment (M5oul) [inline scripts/helper-update-version]: I would not put the script into the `scripts` folder as it is made for the application itself.
I would add this script at the root of this repository.

File: scripts/helper-update-version [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ # Retrieve sources, get sha256sum, parse package version and update these strings in manifest, README and /conf/app.src
+ 
+url=$(curl -s https://api.github.com/repos/duniter/cesium/releases/latest | grep "browser_" | grep "web" | head -1 | cut -d\" -f4)
+wget -nc --quiet $url -P ./tmp
+CHECKSUM=$(sudo sha256sum ./tmp/cesium-*-web.zip | head -c 64)
+sed -i "s/SOURCE_SUM=.*/SOURCE_SUM=${CHECKSUM}/" ../conf/app.src
+
+PACKAGE_VERSION=$(curl -s https://api.github.com/repos/duniter/cesium/releases/latest | grep "tag_name" | head -1 | cut -d\" -f4)
+sed -i "s#SOURCE_URL=.*#SOURCE_URL=https://github.com/duniter/cesium/releases/download/${PACKAGE_VERSION}/cesium-${PACKAGE_VERSION}-web.zip#" ../conf/app.src
```
Comment (M5oul) [inline scripts/helper-update-version]: You are already retrieving this URL upper.

May be you can extract the version from the URL.
I don’t know if it is easier.

File: scripts/helper-update-version [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+sed -i "s#\*\*Shipped version:\*\*.*#\*\*Shipped version:\*\* ${PACKAGE_VERSION}#" ../README.md
+sed -i "s#    \"version\": \".*#    \"version\": \"${PACKAGE_VERSION}\~ynh1\",#" ../manifest.json
+
+sudo rm -f -R ./tmp
+
+git push
+git request-pull
```
Comment (M5oul) [inline scripts/helper-update-version]: You can automatically commit that way:
```bash
git commit README.md manifest.json conf/app.src  -m "v$VERSION"
```

I gave you the rights on this repository.
What are you trying to achieve this this two commands?

Comment (Thatoo) [reply scripts/helper-update-version]: I guess I wanted to do that but as I didn't have the rights, I wanted to do it on my repo and send you a PR at the same time.
Thank you for the rights!
`git commit README.md manifest.json conf/app.src  -m "$PACKAGE_VERSION"`
I changed `"v$VERSION"` to `"$PACKAGE_VERSION"`

File: scripts/helper-update-version (hunk 1 above)
Comment (Thatoo) [reply scripts/helper-update-version]: I guess I don't anymore... It's a remaining of earlier tests...

Comment (Thatoo) [reply scripts/helper-update-version]: Done

File: scripts/helper-update-version (hunk 2 above)
Comment (Thatoo) [reply scripts/helper-update-version]: Done

File: scripts/helper-update-version (hunk 4 above)
Comment (Thatoo) [reply scripts/helper-update-version]: Done

File: scripts/helper-update-version (hunk 3 above)
Comment (Thatoo) [reply scripts/helper-update-version]: Done!

**Comment by Thatoo** [comment]
I guess it's all done. Now I have to check this question of rights.

**Comment by Thatoo** [comment]
@M5oul If you are satisfied with this, I let you merge these two PR and then I'll take care of updating versions.

**Comment by Thatoo** [comment]
Should I not add `git push` at the end of the script?
Then I only need to login and it's all done.
I should check if there is a way to stay logged in from my personal computer.

**Comment by M5oul** [comment]
You do not need to take care of the publication of the commit from within the script. That should be done later by the maintainer.

**Comment by Thatoo** [comment]
OK, understood no publication of the commit from within the script.

I updated. For some reason, a small space was added at the beginning of the script. For me it works.

**Pull request merged**